src/norm/model.nim

```diff
@@ -114,3 +114,25 @@ proc checkRo*(T: typedesc[Model]) =
   when T.hasCustomPragma(ro):
     {.error: "can't use mutating procs with read-only models".}
 
+proc getRelatedFieldNameOn*[T: Model, M: Model](targetModel: typedesc[T], sourceModel: typedesc[M]): string  {.raises: [FieldDefect].} =
+  for sourceFieldName, sourceFieldValue in sourceModel()[].fieldPairs:
+    #Handles case where field is an int64 with fk pragma
+    when sourceFieldValue.hasCustomPragma(fk):
+      when T.table() == sourceFieldValue.getCustomPragmaVal(fk).table():
+        return sourceFieldName
+    
+    #Handles case where field is a Model type
+    elif sourceFieldValue is Model:
+      when T.table() == sourceFieldValue.type().table():
+        return sourceFieldName
+    
+    #Handles case where field is a Option[Model] type
+    elif sourceFieldValue is Option:
+      when sourceFieldValue.get() is Model:
+        when T.table() == genericParams(sourceFieldValue.type()).get(0).table():
+          return sourceFieldName
+
+  raise newException(
+    FieldDefect, 
+    "Tried getting foreign key field from model '" & name(sourceModel.type()) & "' to model '" & name(targetModel.type()) & "' but there is no such field!"
```

Comment:
> Since strutils is already imported, you can replace that with a more readable "$# $#" % [foo, bar] syntax.

Funnily enough, since then I've also gone over to using formatstrings. Do you have a heavy preference for the $# syntax or would using the strformat lib also work for you (should have no performance impact since this all works at compile time)?

Reply:
@moigagoo 
I fixed that and restructured the proc a bit. Previously this would have very poorly understandable behaviour if you had a model A with 2 FK fields to model B. It would secretly hand you the first and ignore the second one. Now there's a way to specify which field name you want to use for the selection. `getRelatedFieldNameTo` is now only used for an optional convenience proc (an overload of `selectOneToMany`) if you don't want to specify that field name and want norm to infer it for you.

I used strformat for this one though as I personally prefer it over strutils formatting syntax.